### PR TITLE
feat: SAML UI: Create updateSAML mutation

### PR DESCRIPTION
**Pull request by gitstart-parabol**
Fixes https://github.com/ParabolInc/parabol/issues/7643

* Create `updateSAML` mutation similar to [enableSAMLForDomain](https://github.com/ParabolInc/parabol/blob/d75533910d56f4358a08f7477fa3df3a72013522/packages/server/graphql/private/mutations/enableSAMLForDomain.ts#L70)
* arguments: `orgId`, `metadata`
* `metadata` can be `null` for disabling SAML again (should also update` url` in the `SAML` table)
* only callable by a `billingLeader` on an `enterprise` organization
* requires an existing `SAML` entry, otherwise fails

 ## Testing scenarios
 * Go to http://localhost:3000/admin/graphql
 * Run the following mutation and update the variables.
```
mutation updateSaml($orgId: ID!, $metadata: String) {
  updateSAML(orgId: $orgId, metadata:$metadata) {
    ... on UpdateSAMLSuccess {
      success
    }
    ... on ErrorPayload {
      error {
        title
        message
      }
    }
  }
}


```

### CODE REVIEW

**Comment by gitstart-parabol** [comment]
@BartoszJarocki this PR is ready for review

**Comment by gitstart-parabol** [comment]
Hi @Dschoordsch, I just wanted to find out how soon will we be getting reviews on this. Thanks.

File: packages/server/graphql/public/typeDefs/updateSAML.graphql [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+"""
+Return value for updateSAML, which could be an error
+"""
+union UpdateSAMLPayload = ErrorPayload | UpdateSAMLSuccess
+
+type UpdateSAMLSuccess {
+  """
+  Describe the first return field here
```
Comment (BartoszJarocki) [inline packages/server/graphql/public/typeDefs/updateSAML.graphql:25]: -1 no generic description please

File: packages/server/graphql/public/typeDefs/updateSAML.graphql [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  """
+  success: Boolean!
```
Comment (BartoszJarocki) [inline packages/server/graphql/public/typeDefs/updateSAML.graphql:27]: @Dschoordsch I'm not 100% sure how this is going to work, do we need to return newly created/updated object id here and be able to resolve it? I'm asking as I'm not sure if frontend needs more data than `true/false`

File: packages/server/graphql/public/mutations/updateSAML.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  const samlRecord = await r
+    .table('SAML')
+    .getAll(orgId, {index: 'orgId'})
+    .limit(1)
+    .nth(0)
+    .default(null)
+    .run()
+
+  if (!samlRecord) {
+    return {error: {message: 'SAML has not been created in Parabol yet.'}}
```
Comment (BartoszJarocki) [inline packages/server/graphql/public/mutations/updateSAML.ts:18]: +1 as this is something called by the user, I'd say we should log it to sentry, check the usage of `standardError` https://github.com/ParabolInc/parabol/blob/a1ca44dda2254fe6747e64e2c358c9dd53f7237b/packages/server/utils/standardError.ts#L3

File: packages/server/graphql/public/mutations/updateSAML.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  }
+
+  const url = metadata ? getSignOnURL(metadata, samlRecord.id) : null
+  if (url instanceof Error) return {error: {message: url.message}}
```
Comment (BartoszJarocki) [inline packages/server/graphql/public/mutations/updateSAML.ts:22]: +1 [`standardError`](https://github.com/ParabolInc/parabol/blob/a1ca44dda2254fe6747e64e2c358c9dd53f7237b/packages/server/utils/standardError.ts#L3)

File: packages/server/graphql/public/mutations/updateSAML.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import getRethink from '../../../database/rethinkDriver'
+import {MutationResolvers} from '../resolverTypes'
+import getSignOnURL from './helpers/SAMLHelpers/getSignOnURL'
+
+const updateSAML: MutationResolvers['updateSAML'] = async (_source, {orgId, metadata}, {}) => {
```
Comment (BartoszJarocki) [inline packages/server/graphql/public/mutations/updateSAML.ts:5]: @Dschoordsch just making sure as I don't see anything subscriptions related, is this mutation supposed to publish any data on `org` level? Or is the UI only for the `billing leader` and there's no need to do anything subscriptions related?

File: codegen.json [hunk 6]
```diff
@@ -69,6 +70,7 @@
           "UpdateDimensionFieldSuccess": "./types/UpdateDimensionFieldSuccess#UpdateDimensionFieldSuccessSource",
           "UpdateGitLabDimensionFieldSuccess": "./types/UpdateGitLabDimensionFieldSuccess#UpdateGitLabDimensionFieldSuccessSource",
           "UpdateMeetingPromptSuccess": "./types/UpdateMeetingPromptSuccess#UpdateMeetingPromptSuccessSource",
+          "UpdateSAMLSuccess": "./types/UpdateSAMLSuccess#UpdateSAMLSuccessSource",
```
Comment (BartoszJarocki) [inline codegen.json:73]: -1 no need to add mapping if there's no such file as graphql codegen will generate incorrect types

**Review by BartoszJarocki — CHANGES_REQUESTED** [review]

File: packages/server/graphql/public/mutations/updateSAML.ts (hunk 5 above)
Comment (Dschoordsch) [reply packages/server/graphql/public/mutations/updateSAML.ts:5]: The UI is only for the billing leader. There could be multiple, but that's an edge case we don't need to fix right now.

Comment (BartoszJarocki) [reply packages/server/graphql/public/mutations/updateSAML.ts:5]: got it

File: packages/server/graphql/public/typeDefs/updateSAML.graphql (hunk 2 above)
Comment (gitstart-parabol) [reply packages/server/graphql/public/typeDefs/updateSAML.graphql:27]: @BartoszJarocki, here is a  [draft](https://github.com/ParabolInc/parabol/pull/7788) for frontend, where we use this `mutation`  for enabling and disabling SSO. we have opted to perform a refetch to inform future decisions based on the query result.

File: codegen.json (hunk 6 above)
Comment (gitstart-parabol) [reply codegen.json:73]: @BartoszJarocki  this was  autogenerated by `codegen` at the point of creating this `mutation`,  I would appreciate your input on whether or not it is necessary to remove it?

**Comment by gitstart-parabol** [comment]
@BartoszJarocki this PR is ready for review.

**Review by BartoszJarocki — APPROVED** [review]

**Review by Dschoordsch — APPROVED** [review]
Works. A could of minor nits which can be done in a follow up.

**Pull request merged**